hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/CommonPathCapabilities.java

```diff
@@ -123,4 +125,10 @@ private CommonPathCapabilities() {
    */
   public static final String FS_XATTRS = "fs.capability.paths.xattrs";
 
+  /**
+   * Experimental support for {@link BatchListingOperations}.
+   */
+  @InterfaceStability.Unstable
+  public static final String FS_EXPERIMENTAL_BATCH_LISTING =
+      "fs.capability.experimental.batch.listing";
```

Comment:
nit: with the key already marked as Unstable, is it necessary to also add "experimental" in the key/value? thinking whether there will be any backward compatibility issue if we decide to make this stable in future and change it to something like `fs.capability.batch.listing`.

Reply:
I'm happy either way. The main justification for separation is if there was ever a transition point where the new builder and the interface here coexisted. If our goal is a single switch, we can go with the final name

Reply:
If you are OK, I'm inclined towards the final name - have seen enough clients in the past that use string value as opposed to the variable name in various places and a switch may just make things silently stop working.

Reply:
ok I will fix that tomorrow. Any other comments? Or are you happy with the patch? Because we cannot ship 3.3.0 with something in FileSystem declaring itself as stable...

Reply:
I think that's all the comments from me. Otherwise LGTM!